src/components/AttendeeForm.tsx

```diff
@@ -0,0 +1,225 @@
+'use client'
+
+import { comment } from 'postcss'
+import type { ChangeEvent } from 'react'
+import { useState } from 'react'
+
+import { usePocketBase } from '../contexts/PocketBaseContext'
+import { LolElo } from '../models/LolElo'
+import { TeamRole } from '../models/TeamRole'
+import { FileInput, Input, SelectInput } from './FormInputs'
+
+type Props = {
+  tournamentId: string
+  onSuscribeOk: () => void
+}
+type Inputs = {
+  riotId: string
+  currentElo: LolElo
+  comment: string
+  role: TeamRole
+  championPool: string
+  birthPlace: string
+  avatar: File | null
+}
+type Errors = Partial<Record<keyof Inputs, string>>
+type Touched = Partial<Record<keyof Inputs, boolean>>
+
+const AttendeeForm: React.FC<Props> = ({ tournamentId, onSuscribeOk }) => {
+  const validate = (newInputs: Inputs): Errors => {
+    const newErrors: Errors = {}
+
+    // RiotId
+    const riotIdRegex = /^.{4,16}#[a-zA-Z0-9]{3,5}$/
+    if (!riotIdRegex.test(newInputs.riotId)) {
+      newErrors.riotId = 'Please enter a valid Riot ID'
+    }
+
+    // Comment
+    if (comment.length > 50) {
+      newErrors.comment = 'Please a comment under 50 characters'
+    }
+
+    // BirthPlace
+    if (newInputs.birthPlace === '') {
+      newErrors.birthPlace = 'Please enter a birthplace'
+    }
+
+    // Avatar
+    if (newInputs.avatar === null) {
+      newErrors.avatar = 'Please choose an avatar'
+    }
+
+    return newErrors
+  }
+
+  const [inputs, setInputs] = useState<Inputs>({
+    riotId: '',
+    currentElo: LolElo.values[0],
+    comment: '',
+    role: TeamRole.values[0],
+    championPool: 'One trick poney',
+    birthPlace: '',
+    avatar: null,
+  })
+  const { pb, user } = usePocketBase()
+
+  const [submitError, setSubmitError] = useState<null | string>(null)
+
+  const errors = validate(inputs)
+
+  const [touched, setTouched] = useState<Touched>({})
+
+  const handleChange = (key: keyof Inputs) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputs({ ...inputs, [key]: event.target.value })
+  }
+  const handleSelectChange =
+    (key: keyof Inputs) => (event: React.ChangeEvent<HTMLSelectElement>) => {
+      setInputs({ ...inputs, [key]: event.target.value })
+    }
+
+  const handleBlur = (key: keyof Inputs) => () => {
+    setTouched({ ...touched, [key]: true })
+  }
+
+  const handleFileChange = () => (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target.files
+    if (files !== null && files.length > 0) {
+      const selected = files[0]
+      setInputs({ ...inputs, avatar: selected })
+    }
+  }
+
+  const showErrorMsg = (key: keyof Inputs): boolean =>
+    errors[key] !== undefined && touched[key] !== undefined
+
+  const handleSubmit = () => (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault()
+
+    let obj = {}
+    for (const cle in inputs) {
+      obj = { ...obj, [cle]: true }
+    }
```

Comment:
Tu pourrais améliorer le typage en faisant : 
```suggestion
    let obj: Touched = {}
    for (const cle in inputs) {
      obj = { ...obj, [cle]: true }
    }
```
mais `cle` n'est quand même pas correctement typé, ça me plait pas :
![image](https://github.com/jean-plank/tournadeck/assets/18679417/83887c48-62bf-4b3a-a19c-6412721dd7fa)
(Je voudrais du `keyof Touched`.)


Pourquoi pas :
```suggestion
    const obj = readonlyRecord
      .keys(inputs)
      .reduce<Touched>((acc, key) => ({ ...acc, [key]: true }), {})
```
?

Reply:
C'est pas un peu overkill

Reply:
C'est plus propre, je trouve.  
J'aurais dû interdire les `for` et `while`.

Reply:
Les restrictions liées au covid 19